modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/jaxrs/message/body/MultipartBodyMessageBodyReader.java

```diff
@@ -143,12 +167,26 @@ public MultipartBody readFrom(
 			binaryFiles, contextResolver::getContext, values);
 	}
 
+	private UploadServletRequest _getUploadServletRequest(
```

Comment:
The only doubt I have is, how to develop this method, recursively or just using a while loop:

```
	private UploadServletRequest _getUploadServletRequest(
		ServletRequest servletRequest) {

		while (servletRequest instanceof ServletRequestWrapper) {
			if (servletRequest instanceof UploadServletRequest) {
				return (UploadServletRequest)servletRequest;
			}

			ServletRequestWrapper servletRequestWrapper =
				(ServletRequestWrapper)servletRequest;

			servletRequest = servletRequestWrapper.getRequest();
		}

		if (servletRequest instanceof UploadServletRequest) {
			return (UploadServletRequest)servletRequest;
		}

		return null;
	}
```

What do you think @4lejandrito ?

Reply:
I think a loop should be more efficient. But I am not 100% sure.